include/turtle/catch.hpp

```diff
@@ -0,0 +1,47 @@
+// http://turtle.sourceforge.net
+//
+// Copyright Mathieu Champlon and Ovanes Markarian 2017
+//
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef MOCK_CATCH_HPP_INCLUDED
+#define MOCK_CATCH_HPP_INCLUDED
+
+#include <catch.hpp>
+
+template< typename Result >
+struct catch_mock_error_policy
+{
+    static Result abort()
+    {
+        FAIL( "Aborted" );
+        throw std::runtime_error( "unreachable" );
```

Comment:
@ovanes here I changed the return Result() to throw an exception, because some types cannot be default constructed and as you mentionned in the first place it's only to quite the compiler anyway!